ydb/library/yql/minikql/comp_nodes/mkql_grace_join.cpp

```diff
@@ -836,15 +836,19 @@ class TGraceJoinSpillingSupportState : public TComputationValue<TGraceJoinSpilli
 void DoCalculateWithSpilling(TComputationContext& ctx) {
     UpdateSpilling();
 
-    if (!HasMemoryForProcessing() && !IsSpillingFinalized) {
-        bool isWaitingForReduce = TryToReduceMemoryAndWait();
-        if (isWaitingForReduce) return;
-    }
-
+    ui64 cntrows = 0;
     while (*HaveMoreLeftRows || *HaveMoreRightRows) {
+        if (0 == (cntrows++ % CheckMemoryRowLimit)) { // checks before first loop and
+                                                      // every CheckMemoryRowLimit rows
+            if (!HasMemoryForProcessing() && !IsSpillingFinalized) {
+                bool isWaitingForReduce = TryToReduceMemoryAndWait();
+                if (isWaitingForReduce) break;
+            }
+        }
         bool isYield = FetchAndPackData(ctx);
-        if (isYield) return;
+        if (isYield) break;
```

Comment:
What is the reason for this change?

Reply:
It helps logging and should not affect result (but, indeed, this is unnecessary clutter)